src/schema/schema_worker.c

```diff
@@ -68,29 +68,87 @@ __schema_tiered_worker(WT_SESSION_IMPL *session, const char *uri,
+    WT_CONNECTION_IMPL *conn;
     WT_DECL_RET;
+    int ingest_ret, stable_ret;
+    const char *ingest_uri, *stable_uri;
+
+    conn = S2C(session);
+    ingest_ret = 0;
 
     WT_RET(__wt_session_get_dhandle(session, uri, NULL, NULL, open_flags));
     WT_LAYERED_TABLE *layered = (WT_LAYERED_TABLE *)session->dhandle;
-    const char *stable_uri = layered->stable_uri;
 
+    ingest_uri = layered->ingest_uri;
+    stable_uri = layered->stable_uri;
+
+    /*
+     * FIXME-WT-15413 - Verify assumes the stable table always exists. However, on followers that
+     * have not yet picked up their first checkpoint, the stable constituent will be missing. We
+     * should handle this transient state by skipping stable verification instead of failing with
+     * ENOENT.
+     */
     WT_ASSERT(session, stable_uri != NULL);
+    WT_ASSERT(session, ingest_uri != NULL);
     WT_ASSERT(session, file_func == __wt_verify);
 
+    /*
+     * Verifying stable tables of layered tables uses the existing verify logic.
+     * Ingest tables, however, require special handling:
+     * - On leader: ingest must always be empty/no-op.
+     * - On followers: FIXME-WT-15047 ingest tables are not checkpointed.
+     */
+
+    /* Verify the stable table of the layered table. */
     WT_WITHOUT_DHANDLE(session,
-      ret = __wt_schema_worker(session, stable_uri, file_func, name_func, cfg, open_flags));
+      stable_ret = __wt_schema_worker(session, stable_uri, file_func, name_func, cfg, open_flags));
+
+    if (stable_ret != 0 && stable_ret != EBUSY)
+        WT_ERR_MSG(session, stable_ret, "Verify (layered): %s stable table verification failed. ",
+          stable_uri);
+
+    /*
+     * Verify the ingest table of the layered table. FIXME-WT-15047: Implement ingest table
+     * verification on followers.
+     */
+    if (conn->layered_table_manager.leader) {
+        /*
+         * On leader, if verifying ingest returns EBUSY, it means ingest is not empty (dirty
+         * content, or open cursors), which is an invalid state.
+         */
+        WT_WITHOUT_DHANDLE(session,
+          ingest_ret =
+            __wt_schema_worker(session, ingest_uri, file_func, name_func, cfg, open_flags));
+
+        /*
+         * FIXME-WT-15433 Add an assertion that verifying the ingest table of the leader never
+         * returns EBUSY.
+         */
+        if (ingest_ret != 0)
+            WT_ERR_MSG(session, ingest_ret,
+              "Verify (layered): %s ingest table verification failed. Ingest must always be empty "
+              "on leader.",
+              ingest_uri);
+    }
+
+err:
+    __wt_verbose_level(session, WT_VERB_VERIFY, WT_VERBOSE_DEBUG_2,
+      "Verify (layered): stable table %s returned %s, ingest table %s returned %s", stable_uri,
+      __wt_wiredtiger_error(stable_ret), ingest_uri, __wt_wiredtiger_error(ingest_ret));
 
     WT_TRET(__wt_session_release_dhandle(session));
+
+    ret = (stable_ret != 0) ? stable_ret : ingest_ret;
+
     return (ret);
```

Comment:
If `stable_ret` contains EBUSY and `ingest_ret` has WT_ERROR we should return WT_ERROR (don't work with the current logic)


```suggestion
    /* Ingest is expected to never return EBUSY so it's enough to check it for 0 only */
    return (ingest_ret != 0) ? ingest_ret : stable_ret;
```

Reply:
Nice catch Ivan! Ret is not wrapped in `()` so I will manually add this

Reply:
done fd93286fa4d8a620adc75f29e824cde734ddf8f6